src/components/views/right_panel/TimelineCard.tsx

```diff
@@ -27,77 +28,178 @@ import { Layout } from '../../../settings/enums/Layout';
+                }, () => {
+                    if (payload.event) {
+                        this.timelinePanelRef.current?.scrollToEventIfNeeded(payload.event.getId());
+                    }
+                });
+                break;
+            default:
+                break;
+        }
+    };
+
+    private onScroll = (): void => {
+        if (this.state.initialEventId && this.state.initialEventHighlighted) {
+            dis.dispatch({
+                action: Action.ViewRoom,
+                room_id: this.props.room.roomId,
+                event_id: this.state.initialEventId,
+                highlighted: false,
+                replyingToEvent: this.state.replyToEvent,
+            });
+        }
+    };
+
     private renderTimelineCardHeader = (): JSX.Element => {
         return <div className="mx_TimelineCard__header">
             <span>{ _t("Chat") }</span>
         </div>;
     };
 
     public render(): JSX.Element {
+        const highlightedEventId = this.state.initialEventHighlighted
+            ? this.state.initialEventId
+            : null;
+
         return (
-            <BaseCard
-                className="mx_ThreadPanel mx_TimelineCard"
-                onClose={this.props.onClose}
-                withoutScrollContainer={true}
-                header={this.renderTimelineCardHeader()}
-            >
-                <TimelinePanel
-                    showReadReceipts={false} // TODO: RR's cause issues with limited horizontal space
-                    manageReadReceipts={true}
-                    manageReadMarkers={false} // No RM support in the TimelineCard
-                    sendReadReceiptOnLoad={true}
-                    timelineSet={this.props.room.getUnfilteredTimelineSet()}
-                    showUrlPreview={true}
-                    layout={Layout.Group}
-                    hideThreadedMessages={false}
-                    hidden={false}
-                    showReactions={true}
-                    className="mx_RoomView_messagePanel mx_GroupLayout"
-                    permalinkCreator={this.props.permalinkCreator}
-                    membersLoaded={true}
-                    editState={this.state.editState}
-                    eventId={this.props.initialEvent?.getId()}
-                    resizeNotifier={this.props.resizeNotifier}
-                />
-
-                <MessageComposer
-                    room={this.props.room}
-                    resizeNotifier={this.props.resizeNotifier}
-                    replyToEvent={this.state.replyToEvent}
-                    permalinkCreator={this.props.permalinkCreator}
-                    e2eStatus={this.props.e2eStatus}
-                    compact={true}
-                />
-            </BaseCard>
+            <RoomContext.Provider value={{
+                ...this.context,
+                timelineRenderingType: this.props.timelineRenderingType || TimelineRenderingType.Room,
```

Comment:
```suggestion
                timelineRenderingType: this.props.timelineRenderingType ?? this.context.timelineRenderingType,
```

IMO would be a saner fallback